### ISSUE TITLE
Accessibility regressions in the new explorer view

### ISSUE THREAD

**Issue by Neurrone** [Microsoft/vscode#67156]
Issue Type: <b>Bug</b>

Tested with the latest version of NVDA on the alpha branch.

1. Whether a node can be expanded is not mentioned. If the item is expanded, there is also no indication of this. So you don't know what is a file or a folder.
2. It counts everything in the tree (including children one level below) when announcing the position of the item in the list. For example, a folder could have its position read as 2 of 18, and a file inside as 3 of 18. I find this confusing, because I expect the count to only include the current item, and other siblings of this node.
3. Trying to type to hopefully use first-letter navigation to move down the list (e.g, pressing e to move to a file that starts with e) doesn't seem to work. This is a massive productivity pain because I'm forced to arrow down the list to find what I want. I'm aware of the option to open a file if you know its name, but it doesn't address the case of wanting to quickly see the directory structure of a sepcific subdir.
4. I've also encountered a bug where arrowing down the list of files in a subfolder isn't accurately reporting the item under the cursor, but is I suspect, reading the siblings after the parent node instead. I'm still not sure how to reproduce it, but using the explorer is impossible when this happens.

VS Code version: Code - Insiders 1.31.0-insider (9fe48f15b1a252a0e9e1ef5975301203baf10b8a, 2019-01-25T08:19:36.277Z)
OS version: Windows_NT x64 10.0.17134

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz (8 x 1992)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|15.86GB (10.21GB free)|
|Process Argv||
|Screen Reader|yes|
|VM|36%|

</details><details><summary>Extensions (7)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-markdownlint|Dav|0.23.0
vscode-eslint|dba|1.8.0
python|ms-|2018.12.1
cpptools|ms-|0.21.0
Go|ms-|0.8.0
rust|rus|0.5.3
markdown-preview-enhanced|shd|0.3.11

</details>

**Comment by isidorn** [comment]
@Neurrone thanks a ton for testing this! We really appreciate it

1. I can reproduce this and we will fix this
2. I can reproduce this and we will fix this
3. Does this get better if you set `"workbench.list.keyboardNavigation": "simple"` setting?. @joaomoreno we should consider to turn this automatically on if we detect that a user has a screen reader attached. The issue here is that the highlight option is probably confusing if there is no visual feedback
4. I can not reproduce this and it would be great if we could find some repro steps

**Comment by joaomoreno** [comment]
> aria-expanded attribute should be added
> posinset should be in a context of the parent, not in the whole list

**Comment by joaomoreno** [comment]
Pushed proper `aria-posinset` and `aria-setsize` attributes. Thanks for this issue @Neurrone, please try (3) again and let us know if you can reproduce (4).

**Comment by Neurrone** [comment]
@isidorn joaomoreno apologies for the late reply.

1. Verified that it is now working properly.
2. This works partially. List item counts when in child nodes are read correctly, but when in level 1 of the tree, child items that become visible after expanding siblings are still part of the count
3. OMG this is amazing; I've wasted time arrowing through the tree because i couldn't use type-ahead navigation. What is the default behaviour like if this setting is not enabled? Please also consider setting this automatically when a screen reader is detected, and highlighting that this option exists, in the upcoming release notes, because it was a major pain point.
4. No luck reproducing this so far. I'll see if I can get this to happen browsing in Vs Code's repo.

**Comment by isidorn** [comment]
@Neurrone thanks for verifying 1
2. @joaomoreno can investigate more

Regarding 3. yes we shuold auto switch this when we detect screen reader. I have created a feature request for that for the next milestone https://github.com/Microsoft/vscode/issues/67744

Thanks again for giving us great feedback and let us know of other pain points you are experiencing.

**Comment by joaomoreno** [comment]
Can you elaborate (2)? I did not understand. I just reviewed the `aria-posinset` and `aria-setsize` attributes of all elements when filtering in and out and I hey all seem OK to me.

**Comment by Neurrone** [comment]
@joaomoreno For 2, replicate the following folder structure, which contains a single file, and a subfolder which in turn has a file:

* subfolder1
	* file2
* file1

When the subfolder node is collapsed, moving to the file announces its position as 2 of 2, as expected. 

Next, expand the subfolder node. File2's position is correctly announced as level 2, 1 of 1. However, when moving back down to file1, it announces 3 of 3, instead of 2 of 2.

**Comment by Neurrone** [comment]
@joaomoreno 

For 4, open the Vs Code source repository. navigate and expand the src/vs/base folder. When pressing down arrow to read its contents, it announces the sibling folders (code, editor etc) instead of the children of base. This seems to always happen for level 4 and deeper items.

Some other minor feedback:

* The arrow keys wrap around (pressing down from the bottom moves up to the top and vice versa). I'm finding this slightly surprisingly, is this supposed to happen or is there a way for me to turn it off?
* It announces the following when focus enters the explorer tree widget: `Files Explorer. Use the navigation keys to navigate.  tree view`. I feel that the help text about "using the navigation keys to navigate" is redundant, because it doesn't add any additional information. This is pretty minor though in the grand scheme of things.

**Comment by joaomoreno** [comment]
> For 2, replicate the following folder structure, which contains a single file, and a subfolder which in turn has a file:

Great catch! Pushed https://github.com/Microsoft/vscode/commit/b95e51b7979841c992da64ffa7738025f8180695

> For 4, open the Vs Code source repository. navigate and expand the src/vs/base folder. When pressing down arrow to read its contents, it announces the sibling folders (code, editor etc) instead of the children of base. This seems to always happen for level 4 and deeper items.

Yeah I can repro this too: created https://github.com/Microsoft/vscode/issues/68240

> The arrow keys wrap around (pressing down from the bottom moves up to the top and vice versa). I'm finding this slightly surprisingly, is this supposed to happen or is there a way for me to turn it off?

No current option, we can add one if you think it's important.

> * It announces the following when focus enters the explorer tree widget: `Files Explorer. Use the navigation keys to navigate.  tree view`. I feel that the help text about "using the navigation keys to navigate" is redundant, because it doesn't add any additional information. This is pretty minor though in the grand scheme of things.

@isidorn Leaving this to you.

---

@Neurrone Overall, great issues, thanks so much for your passion!

**Comment by Neurrone** [comment]
@joaomoreno thanks to you and the Vs Code team for fixing these so promptly; its rare for me to ever see this happen. I use this as my primary editor at work, so I'm highly motivated to report issues that limit my productivity. Particularly, the problem with 4th level items and deeper not being read is really hard for me to work around.

For the last 2 issues about the help text and wrapping around, they are really not that important. In particular for wrapping, there might be a use case that I haven't thought of.

**Comment by isidorn** [comment]
@Neurrone thank you also from me for nicely testing and providing great suggestions.
Regarding "Files Explorer. Use the navigation keys to navigate." What would you prefer, simply "Files Explorer"?

Also in the future when filling any issues to vscode feel free to ping me directly on them.

**Comment by Neurrone** [comment]
Thanks @isidorn. Lets leave it the way it is now, unless there is similar feedback about this from others.
